src/modules/processing/services/snapchat.js

```diff
@@ -0,0 +1,89 @@
+import { genericUserAgent } from "../../config.js";
+import { getRedirectingURL } from "../../sub/utils.js";
+
+const SPOTLIGHT_VIDEO_REGEX = /<link data-react-helmet="true" rel="preload" href="(https:\/\/cf-st\.sc-cdn\.net\/d\/[\w.?=]+&amp;uc=\d+)" as="video"\/>/;
+const NEXT_DATA_REGEX = /<script id="__NEXT_DATA__" type="application\/json">({.+})<\/script><\/body><\/html>$/;
+
+async function getSpotlight(pathname) {
```

Comment:
the interface for this function would ideally be `getSpotlight(spotlightId)`, instead of passing in the path (which is a lot harder to decipher) you just pass the ID (and you already know the rest of the path anyways)..

similarly for getStory() below except with username + storyId